ksqldb-common/src/main/java/io/confluent/ksql/util/QueryApplicationId.java

```diff
@@ -50,6 +50,6 @@ public static String build(
   }
 
   private static String addTimeSuffix(final String original) {
-    return String.format("%s_%d", original, System.currentTimeMillis());
+    return String.format("%s_%d", original, System.currentTimeMillis()).replace("-","_");
```

Comment:
looking at this code I don't think this is the right way to make this change for a few reasons:

1. we're not respecting configurations that are possibly passed in by the user (e.g. `KSQL_TRANSIENT_QUERY_NAME_PREFIX_CONFIG`) - we shouldn't just be replacing chars in their passed in config
2. it doesn't really make sense for the `addTimeSuffix` method to be doing this

Instead, I think we should look for a way that solves the underlying problem instead of adding shim to fix a bad queryID after it's created. Namely, the underlying problem is that the default generation for transient query ids uses the `_` character. How can we fix that? (hint: maybe the queryID and the applicationID don't necessarily need to be the same - which they are not the same for persistent queries)